pkg/kube/clientset.go

```diff
@@ -25,12 +25,30 @@ import (
 )
 
 // NewClientSet returns a new Kubernetes clientset
-func NewClientSet(kubeContext, kubeConfig string) (*kubernetes.Clientset, error) {
+func NewClientSet(kubeContext, kubeConfig string, impersonateUser string, impersonateGroup string) (*kubernetes.Clientset, error) {
 	config, err := getKubeConfig(kubeContext, kubeConfig)
 	if err != nil {
 		return nil, err
 	}
-
+	groupArray := []string {impersonateGroup}
+	// --as and --as-group
+	if impersonateUser != "" && impersonateGroup != "" {
+		config.Impersonate = rest.ImpersonationConfig {
+        	UserName: impersonateUser,
+			Groups: groupArray,
+		}
+	// --as
+	} else if impersonateUser != "" && impersonateGroup == "" {
+		config.Impersonate = rest.ImpersonationConfig {
+			UserName: impersonateUser,
+		}
+	// --as-group
+	} else if impersonateUser == "" && impersonateGroup != "" {
+		config.Impersonate = rest.ImpersonationConfig {
+			Groups: groupArray,
+		}
+	}
```

Comment:
Maybe something like this would be a bit simpler?

```suggestion
	if impersonateUser != "" || impersonateGroup != "" {
		config.Impersonate = rest.ImpersonationConfig{}
		if impersonateUser != "" {
			config.Impersonate.UserName = impersonateUser
		}
		if impersonateGroup != "" {
			config.Impersonate.Groups = []string{impersonateGroup}
		}
	}
```

Reply:
I love it! I've amended my commit and tested each scenario. Thanks for the suggestion.